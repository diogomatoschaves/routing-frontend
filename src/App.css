@font-face {
  font-family: "BasisGrotesque Light";
  src: url(./utils/assets/fonts/basisGrotesque/BasisGrotesqueTrial-Light.otf) format("opentype");
}

@font-face {
  font-family: "BasisGrotesque Medium";
  src: url(./utils/assets/fonts/basisGrotesque/BasisGrotesqueTrial-Medium.otf) format("opentype");
}

@font-face {
  font-family: "BasisGrotesque Bold";
  src: url(./utils/assets/fonts/basisGrotesque/BasisGrotesqueTrial-Bold.otf) format("opentype");
}

body {
  font-size: 20px;
  font-family: "BasisGrotesque Medium", Lato,'Helvetica Neue',Arial,Helvetica,sans-serif !important;
}

body * :not(.icon){
  font-family: "BasisGrotesque Medium", Lato,'Helvetica Neue',Arial,Helvetica,sans-serif !important;
}


.custom-font-medium {
  font-family: "BasisGrotesque Medium", Lato,'Helvetica Neue',Arial,Helvetica,sans-serif;
}

.custom-font-bold {
  font-family: "BasisGrotesque Bold", Lato,'Helvetica Neue',Arial,Helvetica,sans-serif;
}

.custom-marker.start {
  font-size: 3.6em !important;
  color: #66c2a5;
}

.custom-marker.end {
  font-size: 3.6em !important;
  color: #d53e4f;
}

.custom-marker.waypoint {
  font-size: 2.7em !important;
  color: #007a93;
}

.key {
  color: red;
}

.number {
  color: blue;
}