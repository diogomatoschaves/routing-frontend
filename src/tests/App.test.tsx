import React from 'react'
import { MemoryRouter } from 'react-router-dom'
import TestRenderer from 'react-test-renderer'
import App from '../components/App'

const mockMatch = {
  isExact: true,
  params: {
    profile: 'car',
    start: '52.5332192, 13.3329111',
    end: '52.5160144, 13.3820557'
  },
  path: '/:profile/:start/:end',
  url: '/car/52.5332192, 13.3329111/52.5160144, 13.3820557'
}

it('expect to render App component', () => {
  const testInstance = TestRenderer.create(
    <MemoryRouter initialEntries={['/']}>
      <App match={mockMatch} />
    </MemoryRouter>
  )
  expect(testInstance.toJSON()).toMatchSnapshot()
})
