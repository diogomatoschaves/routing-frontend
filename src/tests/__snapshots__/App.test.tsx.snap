// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render App component 1`] = `
<div
  className="sc-gPEVay bJvjiw"
>
  <div
    className="ui segment pushable sc-iRbamj gOGLLM"
  >
    <div
      className="sc-bwzfXH sc-jlyJG hRSjkk"
      direction="row"
      width="31%"
    >
      <div
        className="sc-ifAKCX sc-gipzik dvMpeQ"
        height="10%"
        width={0}
      />
      <button
        className="ui icon button options-button sc-csuQGl dIyLUF"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        padding="10px 4px"
      >
        <i
          aria-hidden="true"
          className="chevron right big icon scale visible transition"
          onClick={[Function]}
          style={
            Object {
              "animationDuration": undefined,
            }
          }
        />
      </button>
    </div>
    <div
      className="ui vertical labeled icon ui overlay left very wide sidebar menu"
    >
      <div
        className="sc-bwzfXH dlzeER"
      >
        <div
          className="sc-bwzfXH cwyxvS"
          direction="row"
        >
          <div
            className="sc-bwzfXH lacXsB"
            direction="row"
          >
            <div
              className="sc-bwzfXH sc-cSHVUG ddeyVz"
              color={true}
              id="default"
              onClick={[Function]}
              width="100px"
            >
              Interactive
            </div>
            <div
              className="sc-bwzfXH sc-cSHVUG hwPHJs"
              color={false}
              id="debug"
              onClick={[Function]}
              width="100px"
            >
              Debugging
            </div>
          </div>
        </div>
        <div
          className="ui divider sc-htoDjs jHFehN"
          width="90%"
        />
        <div
          className="sc-bwzfXH dpPFld"
          direction="row"
          height="70px"
        >
          <div
            className="sc-bwzfXH jzFvfk"
            width="40%"
          >
            <div
              className="sc-bwzfXH dfIKGx"
              direction="row"
              height="45px"
            >
              <div
                className="sc-bwzfXH gmInAj"
                onClick={[Function]}
                width="10%"
              >
                <i
                  aria-hidden="true"
                  className="grey angle left large icon sc-jTzLTM kMqZix"
                  onClick={[Function]}
                />
              </div>
              <div
                className="sc-bwzfXH gCPlrj"
                width="60%"
              >
                <div
                  className="ui large label sc-fjdhpX fsHjxY"
                  onClick={[Function]}
                >
                  No Traffic
                </div>
              </div>
              <div
                className="sc-bwzfXH gmInAj"
                onClick={[Function]}
                width="10%"
              >
                <i
                  aria-hidden="true"
                  className="grey angle right large icon sc-jTzLTM kMqZix"
                  onClick={[Function]}
                />
              </div>
            </div>
          </div>
        </div>
        <div
          className="sc-bwzfXH kwvKmS"
          width="80%"
        >
          <div
            className="ui header sc-dnqmqq eRsqKH"
            overridecolor="#00677f"
          >
            Request Endpoint
          </div>
          <div
            className="sc-bwzfXH iSONxz"
            direction="row"
          >
            <div
              aria-expanded={false}
              className="ui fluid search selection dropdown sc-bZQynM gFXwrf"
              onBlur={[Function]}
              onChange={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onMouseDown={[Function]}
              role="combobox"
            >
              <input
                aria-autocomplete="list"
                autoComplete="off"
                className="search"
                onChange={[Function]}
                style={
                  Object {
                    "width": undefined,
                  }
                }
                tabIndex={0}
                type="text"
                value=""
              />
              <div
                aria-live="polite"
                className="text"
                role="alert"
              >
                https://routing.develop.otonomousmobility.com/\${PROFILE}
              </div>
              <i
                aria-hidden="true"
                className="dropdown icon"
                onClick={[Function]}
              />
              <div
                className="menu transition"
                role="listbox"
              >
                <div
                  aria-checked={true}
                  aria-selected={true}
                  className="active selected item"
                  onClick={[Function]}
                  role="option"
                  style={
                    Object {
                      "pointerEvents": "all",
                    }
                  }
                >
                  <span
                    className="text"
                  >
                    https://routing.develop.otonomousmobility.com/\${PROFILE}
                  </span>
                </div>
                <div
                  aria-checked={false}
                  aria-selected={false}
                  className="item"
                  onClick={[Function]}
                  role="option"
                  style={
                    Object {
                      "pointerEvents": "all",
                    }
                  }
                >
                  <span
                    className="text"
                  >
                    https://routing.staging.otonomousmobility.com/\${PROFILE}
                  </span>
                </div>
                <div
                  aria-checked={false}
                  aria-selected={false}
                  className="item"
                  onClick={[Function]}
                  role="option"
                  style={
                    Object {
                      "pointerEvents": "all",
                    }
                  }
                >
                  <span
                    className="text"
                  >
                    https://routing.testing.otonomousmobility.com/\${PROFILE}
                  </span>
                </div>
                <div
                  aria-checked={false}
                  aria-selected={false}
                  className="item"
                  onClick={[Function]}
                  role="option"
                  style={
                    Object {
                      "pointerEvents": "all",
                    }
                  }
                >
                  <span
                    className="text"
                  >
                    http://localhost:5000
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="sc-bwzfXH kwvKmS"
          width="80%"
        >
          <div
            className="sc-bwzfXH ewJcDl"
            direction="row"
          >
            <div
              className="ui header sc-dnqmqq eRsqKH"
              overridecolor="#00677f"
            >
              Request Body
            </div>
            <button
              className="ui button body-textarea sc-jAaTju dPvgCX"
              id="bodyEdit"
              marginleft="10px"
              onClick={[Function]}
            >
              Edit
            </button>
          </div>
          <div
            className="sc-bwzfXH ewJcDl"
            direction="row"
            width="100%"
          >
            <div
              className="sc-chPdSV jbjPGB"
            >
              <div
                className="sc-kAzzGY hUkmXL"
                onClick={[Function]}
              >
                
                <i
                  aria-hidden="true"
                  className="caret down icon"
                  onClick={[Function]}
                />
              </div>
              <div
                className="sc-chPdSV jbjPGB"
              >
                <div
                  className="sc-kAzzGY hUkmXL"
                  onClick={[Function]}
                >
                  locations
                  <i
                    aria-hidden="true"
                    className="caret down icon"
                    onClick={[Function]}
                  />
                </div>
                empty array
              </div>
              <div
                className="sc-chPdSV jbjPGB"
              >
                <span
                  className="sc-kgoBCf jDyuzH"
                >
                  reportGeometry
                  : 
                </span>
                <span
                  className="sc-kGXeez hOygGA"
                >
                  true
                </span>
              </div>
            </div>
          </div>
          <div
            className="sc-brqgnP izYMgW header"
          >
            Edit Request Body
          </div>
          <div
            className="sc-cMljjf hjreQu content"
          >
            <form
              className="ui form sc-jzJRlG kEUzBi"
              color="rgb(100, 100, 100)"
              focuscolor="#5097ab"
              onSubmit={[Function]}
            >
              <textarea
                className="textarea"
                id="body"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onInput={[Function]}
                placeholder=""
                rows={15}
                value="{
  \\"locations\\": [],
  \\"reportGeometry\\": true
}"
              />
            </form>
          </div>
          <div
            className="actions"
          >
            <div
              className="sc-bwzfXH fcyEOL"
              direction="row"
            >
              <button
                alignend={true}
                backgroundcolor="#00677f"
                className="ui button sc-gzVnrw cpzMIe"
                onClick={[Function]}
              >
                Confirm
              </button>
            </div>
          </div>
        </div>
        <div
          className="sc-bwzfXH kwvKmS"
          width="80%"
        >
          <div
            className="sc-bwzfXH ewJcDl"
            direction="row"
          >
            <div
              className="ui header sc-dnqmqq eRsqKH"
              overridecolor="#00677f"
            >
              Response
            </div>
            <button
              className="ui button response-textarea sc-jAaTju dPvgCX"
              id="responseEdit"
              marginleft="10px"
              onClick={[Function]}
            >
              View source
            </button>
          </div>
          <div
            className="sc-bwzfXH ewJcDl"
            direction="row"
            width="100%"
          >
            <div
              className="sc-chPdSV jbjPGB"
            >
              <div
                className="sc-kAzzGY hUkmXL"
                onClick={[Function]}
              >
                
                <i
                  aria-hidden="true"
                  className="caret down icon"
                  onClick={[Function]}
                />
              </div>
              <div
                className="sc-chPdSV jbjPGB"
              >
                <span
                  className="sc-kgoBCf jDyuzH"
                >
                  code
                  : 
                </span>
                <span
                  className="sc-kGXeez hOygGA"
                >
                  Ok
                </span>
              </div>
              <div
                className="sc-chPdSV jbjPGB"
              >
                <div
                  className="sc-kAzzGY hUkmXL"
                  onClick={[Function]}
                >
                  routes
                  <i
                    aria-hidden="true"
                    className="caret down icon"
                    onClick={[Function]}
                  />
                </div>
                <div
                  className="sc-chPdSV jbjPGB"
                >
                  <div
                    className="sc-kAzzGY hUkmXL"
                    onClick={[Function]}
                  >
                    0
                    <i
                      aria-hidden="true"
                      className="caret down icon"
                      onClick={[Function]}
                    />
                  </div>
                  <div
                    className="sc-chPdSV jbjPGB"
                  >
                    <span
                      className="sc-kgoBCf jDyuzH"
                    >
                      totalDuration
                      : 
                    </span>
                    <span
                      className="sc-kGXeez hOygGA"
                    >
                      0
                    </span>
                  </div>
                  <div
                    className="sc-chPdSV jbjPGB"
                  >
                    <span
                      className="sc-kgoBCf jDyuzH"
                    >
                      totalDistance
                      : 
                    </span>
                    <span
                      className="sc-kGXeez hOygGA"
                    >
                      0
                    </span>
                  </div>
                  <div
                    className="sc-chPdSV jbjPGB"
                  >
                    <div
                      className="sc-kAzzGY hUkmXL"
                      onClick={[Function]}
                    >
                      legs
                      <i
                        aria-hidden="true"
                        className="caret down icon"
                        onClick={[Function]}
                      />
                    </div>
                    <div
                      className="sc-chPdSV jbjPGB"
                    >
                      <div
                        className="sc-kAzzGY hUkmXL"
                        onClick={[Function]}
                      >
                        0
                        <i
                          aria-hidden="true"
                          className="caret down icon"
                          onClick={[Function]}
                        />
                      </div>
                      <div
                        className="sc-chPdSV jbjPGB"
                      >
                        <div
                          className="sc-kAzzGY hUkmXL"
                          onClick={[Function]}
                        >
                          geometry
                          <i
                            aria-hidden="true"
                            className="caret down icon"
                            onClick={[Function]}
                          />
                        </div>
                        empty array
                      </div>
                      <div
                        className="sc-chPdSV jbjPGB"
                      >
                        <span
                          className="sc-kgoBCf jDyuzH"
                        >
                          duration
                          : 
                        </span>
                        <span
                          className="sc-kGXeez hOygGA"
                        >
                          0
                        </span>
                      </div>
                      <div
                        className="sc-chPdSV jbjPGB"
                      >
                        <span
                          className="sc-kgoBCf jDyuzH"
                        >
                          distance
                          : 
                        </span>
                        <span
                          className="sc-kGXeez hOygGA"
                        >
                          0
                        </span>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                className="sc-chPdSV jbjPGB"
              >
                <div
                  className="sc-kAzzGY hUkmXL"
                  onClick={[Function]}
                >
                  locations
                  <i
                    aria-hidden="true"
                    className="caret down icon"
                    onClick={[Function]}
                  />
                </div>
                empty array
              </div>
            </div>
          </div>
          <div
            className="sc-brqgnP izYMgW header"
          >
            View Response
          </div>
          <div
            className="sc-cMljjf hjreQu content"
          >
            <form
              className="ui form sc-jzJRlG kEUzBi"
              color="rgb(100, 100, 100)"
              focuscolor="#5097ab"
              onSubmit={[Function]}
            >
              <textarea
                className="textarea"
                id="response"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onInput={[Function]}
                placeholder=""
                rows={15}
                value="{
  \\"code\\": \\"Ok\\",
  \\"routes\\": [
    {
      \\"totalDuration\\": 0,
      \\"totalDistance\\": 0,
      \\"legs\\": [
        {
          \\"geometry\\": [],
          \\"duration\\": 0,
          \\"distance\\": 0
        }
      ]
    }
  ],
  \\"locations\\": []
}"
              />
            </form>
          </div>
          <div
            className="actions"
          >
            <div
              className="sc-bwzfXH fcyEOL"
              direction="row"
            >
              <button
                alignend={true}
                backgroundcolor="#00677f"
                className="ui button sc-gzVnrw cpzMIe"
                onClick={[Function]}
              >
                Close
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="pusher"
    >
      0
      <div
        className="sc-bwzfXH sc-bwzfXH sc-kkGfuU eDudgG sc-kkGfuU eDudgG"
        direction="column"
      >
        <div
          className="sc-bwzfXH jzxLsv"
          direction="row"
        >
          <div
            className="sc-bwzfXH ewJcDl"
            direction="row"
          >
            <div
              className="sc-bwzfXH sc-cSHVUG ddeyVz"
              color={true}
              id="default"
              onClick={[Function]}
              width="100px"
            >
              Interactive
            </div>
            <div
              className="sc-bwzfXH sc-cSHVUG hwPHJs"
              color={false}
              id="debug"
              onClick={[Function]}
              width="100px"
            >
              Debugging
            </div>
          </div>
        </div>
        <div
          className="sc-bwzfXH islmOK"
          direction="column"
        >
          <div
            className="sc-bwzfXH iSONxz"
            direction="row"
          >
            <div
              id="car-profile"
              onClick={[Function]}
            >
              <div
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
              >
                <div
                  className="sc-htpNat oWmeC"
                  color="#e1e1e1"
                  cursor="pointer"
                >
                  <div
                    className="sc-bwzfXH dmJOrF"
                    height="100%"
                  >
                    <i
                      aria-hidden="true"
                      className="car icon sc-bxivhb vcXGv"
                      fontSize="25px"
                      height="20px"
                      onClick={[Function]}
                      overridecolor="#007a93"
                      padding="0 0 0 0"
                      position="absolute"
                    />
                  </div>
                </div>
              </div>
            </div>
            <div
              id="foot-profile"
              onClick={[Function]}
            >
              <div
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
              >
                <div
                  className="sc-htpNat gQmiyd"
                  color="transparent"
                  cursor="pointer"
                >
                  <div
                    className="sc-bwzfXH dmJOrF"
                    height="100%"
                  >
                    <i
                      aria-hidden="true"
                      className="male icon sc-bxivhb hFCAaS"
                      fontSize="25px"
                      height="20px"
                      onClick={[Function]}
                      overridecolor="#a6cad8"
                      padding="0 0 0 0"
                      position="absolute"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            className="sc-bwzfXH fOQsml"
            direction="row"
          >
            <div
              className="sc-htpNat aNfhF"
              color="#007a93"
            >
              <div
                className="sc-bwzfXH gzsoVo"
                height="100%"
              >
                <i
                  aria-hidden="true"
                  className="map marker alternate icon sc-bxivhb gUqPoa"
                  fontSize="25px"
                  height="20px"
                  onClick={[Function]}
                  overridecolor="white"
                  padding="0 0 0 0"
                  position="absolute"
                />
              </div>
            </div>
            <div
              className="ui fluid input sc-iwsKbI iDHUoR"
            >
              <input
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                placeholder="Origin"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            className="sc-bwzfXH fOQsml"
            direction="row"
          >
            <div
              className="sc-htpNat aNfhF"
              color="#007a93"
            >
              <div
                className="sc-bwzfXH gzsoVo"
                height="100%"
              >
                <i
                  aria-hidden="true"
                  className="map marker icon sc-bxivhb gUqPoa"
                  fontSize="25px"
                  height="20px"
                  onClick={[Function]}
                  overridecolor="white"
                  padding="0 0 0 0"
                  position="absolute"
                />
              </div>
            </div>
            <div
              className="ui fluid input sc-iwsKbI iDHUoR"
            >
              <input
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                placeholder="Destination"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            className="sc-bwzfXH fOQsml"
            direction="row"
          >
            <div
              className="sc-bwzfXH eoImHn"
              direction="row"
              height="50px"
              width="65%"
            >
              <div
                className="ui fitted toggle checkbox custom-class sc-VigVT jawxJK"
                onChange={[Function]}
                onClick={[Function]}
                onMouseDown={[Function]}
                onMouseUp={[Function]}
              >
                <input
                  checked={false}
                  className="hidden"
                  id="googleMapsOption"
                  readOnly={true}
                  tabIndex={0}
                  type="checkbox"
                />
                <label
                  htmlFor="googleMapsOption"
                />
              </div>
              <label
                className="sc-gqjmRU esqmUJ"
              >
                Google Maps
              </label>
            </div>
            <div
              className="sc-bwzfXH cQpeMM"
              direction="row"
              height="50px"
              width="35%"
            >
              <div
                className="ui fitted toggle checkbox custom-class sc-VigVT jawxJK"
                onChange={[Function]}
                onClick={[Function]}
                onMouseDown={[Function]}
                onMouseUp={[Function]}
              >
                <input
                  checked={false}
                  className="hidden"
                  id="trafficOption"
                  readOnly={true}
                  tabIndex={0}
                  type="checkbox"
                />
                <label
                  htmlFor="trafficOption"
                />
              </div>
              <label
                className="sc-gqjmRU esqmUJ"
              >
                Traffic
              </label>
            </div>
          </div>
        </div>
      </div>
      <div
        className="sc-bdVaJa jthvZh"
      />
    </div>
  </div>
</div>
`;
